### PR TITLE
MiniQtHelper: upgrade ComboBox widget

### PR DESCRIPTION
**Pull request by apwojcik**
Added interface functions:
    - GetComboCount() the function returns the number of
      combo box options currently set,
    - SelectComboOption() - selects an active combo box
      option by name from the list of set options.

### CODE REVIEW

File: qrenderdoc/Code/Interface/Extensions.h [hunk 1]
```diff
@@ -908,7 +908,24 @@ passed nothing will happen.
 :param QWidget combo: The combo box.
 :param List[str] options: The new options for the combo box.
 )");
-  virtual void SetComboOptions(QWidget *combo, const rdcarray<rdcstr> &options) = 0;
+  virtual void SetComboBoxOptions(QWidget *combo, const rdcarray<rdcstr> &options) = 0;
```
Comment (baldurk) [inline qrenderdoc/Code/Interface/Extensions.h]: I don't generally promise compatibility on the python APIs, but I don't think breaking source compatibiltiy like this is justified for an arguably marginal at best improvement in consistency.

You could either leave the function as is and name the new functions `GetComboCount` and `SelectComboOption` or name the new functions as you have done and accept the minor difference.

File: qrenderdoc/Code/Interface/Extensions.h [hunk 2]
```diff
@@ -908,7 +908,24 @@ passed nothing will happen.
...
+
+  DOCUMENT(R"(Get the number of options in a drop-down combo box. If another type of widget is
+passed ``0`` will be returned.
```
Comment (baldurk) [inline qrenderdoc/Code/Interface/Extensions.h:914]: Minor note, this is not true as it looks like the implementation (as well as the existing implementation for `SetComboOptions` which has a bug) will crash if another widget is passed. The function needs to check that the return of `qobject_cast` is non-NULL.

**Review by baldurk — CHANGES_REQUESTED** [review]

File: qrenderdoc/Code/Interface/Extensions.h (hunk 1 above)
Comment (apwojcik) [reply qrenderdoc/Code/Interface/Extensions.h]: I will revert the name to the previous convention and change the naming of the new functions.

File: qrenderdoc/Code/Interface/Extensions.h (hunk 2 above)
Comment (apwojcik) [reply qrenderdoc/Code/Interface/Extensions.h:914]: That is fixed. Thanks for pointing this out.

File: qrenderdoc/Code/Interface/Extensions.h [hunk 3]
```diff
@@ -891,7 +891,7 @@ happen.
   DOCUMENT(R"(Create a drop-down combo box widget.
 
 When created there are no pre-defined entries in the drop-down section. This can be changed with
-:meth:`SetComboOptions`.
+:meth:`SetComboBoxOptions`.
```
Comment (baldurk) [inline qrenderdoc/Code/Interface/Extensions.h]: This comment has still been changed (now incorrectly) in your commit.

**Comment by baldurk** [comment]
btw I don't know why you're adding a comment to each of your commits with your email address, but it's not necessary. Git already stores your email address in the metadata of each commit which shows who authored it.

**Comment by apwojcik** [comment]
I have an alias set for `git commit -s ...` that adds `signed-off-by` automatically (as well as trailers `reviewed-by` and `tested-by`). I contribute to other projects requiring such information to be present. I will remove them here.

**Comment by baldurk** [comment]
That's fine you don't have to remove them, I just wasn't aware of that option and I didn't know what it meant so I thought it was something you were adding manually to sign your commits.

File: qrenderdoc/Code/Interface/Extensions.h (hunk 3 above)
Comment (apwojcik) [reply qrenderdoc/Code/Interface/Extensions.h]: Done

**Pull request merged**